src/electron/device-connect-view/components/window-footer.tsx

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { css } from '@uifabric/utilities';
+import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import * as React from 'react';
+
+export interface WindowFooterProps {
+    cancelClick: () => void;
+}
+
+export class WindowFooter extends React.Component<WindowFooterProps> {
```

Comment:
You can make this a pure component rather than a class, since the only thing that this class does is `render()`.
e.g: look at  [blocking-dialog.tsx](https://github.com/microsoft/accessibility-insights-web/blob/master/src/common/components/blocking-dialog.tsx) as an example

Reply:
Changed this and `window-header`. Do you know of any good reading/guidance on the benefits/trade offs of pure components vs new component classes?

Reply:
I just figured out that these are called `function components` rather than `pure components`. I apologize for my mistake in my first comment there. The example of `blocking-dialog` is also not a pure component but a function component.
That said, `Pure Components` is another way of defining a React component. Read more about that at: https://reactjs.org/docs/react-api.html#reactpurecomponent.

Internally, React treats `class component` and `function component` as same. In previous versions of react, the react stateless components (or function component) were seen as pure but that has since been changed by React.

For reading about what kind of advantages it provides, you can take a look at: https://hackernoon.com/react-stateless-functional-components-nine-wins-you-might-have-overlooked-997b0d933dbc?source=post_page-----231e3fbd7108----------------------

Please let me know if you want to discuss this further. I will also put in something for our engineering discussion next Thursday.

Reply:
Thanks!